File: 0567 - Permutation in String/solution.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        int n = s1.size(), val = 0, t[26] = {};
        for(auto i : s1) if(t[i-'a']++ == 0) val++;
        for(int i = 0; i < s2.size(); i++){
            if(i>=n && t[s2[i-n]-'a']++==0) val++;
            if(--t[s2[i]-'a'] == 0) val--;
            if(val == 0) return true;
        }
        return false;
    }
};

File: 0567 - Permutation in String/solution.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        t1, t2, n = [0]*26, [0]*26, len(s1)
        for i in s1:
            t1[ord(i)-ord('a')] += 1
        for i in range(len(s2)):
            if i>=n:
                t2[ord(s2[i-n])-ord('a')] -= 1
            t2[ord(s2[i])-ord('a')] += 1
            if t1==t2:
                return True
        return False